### PR TITLE
MultiOutputRegressor that handles validation data in fit()

### PR DESCRIPTION
**Pull request by brunnedu**
Addresses #1144 

### Summary

This PR solves an issue where validation data that was passed to the `LightGBMModel` or the `CatBoostModel` in `fit()` wasn't handled correctly by the original sklearn `MultiOutputRegressor`.

The solution is a slightly modified `MultiOutputRegressor` class that slices the validation data together with the training data.

- Added `MultiOutputRegressor` in `darts.utils.multioutput` that inherits from sklearn `MultiOutputRegressor` and overwrites its `fit()` method
- `MultiOutpuRegressor.fit()` now slices validation data together with training data as long as it is passed in the parameter `eval_set` as part of `**fit_params`
- Added unit tests that check for this case

### CODE REVIEW

**Comment by brunnedu** [comment]
I also wanted to add some tests for the new `MultiOutputRegressor` class but not sure when I'll get to it. For now, I just added a new test in `test_regression_models.py`.

File: darts/utils/multioutput.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+from joblib import Parallel
+from sklearn.base import is_classifier
+from sklearn.multioutput import MultiOutputRegressor as sk_MultiOutputRegressor
+from sklearn.multioutput import _fit_estimator
+from sklearn.utils.fixes import delayed
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.utils.validation import _check_fit_params, has_fit_parameter
+
+
+class MultiOutputRegressor(sk_MultiOutputRegressor):
```
Comment (hrzn) [inline darts/utils/multioutput.py:10]: Looks good! I wonder if that could be of any use to propose this as a PR to modify the `_MultiOutputEstimator` class in sklearn directly. WDYT?

**Review by hrzn — APPROVED** [review]
Thanks, looks great!

File: darts/utils/multioutput.py (hunk 1 above)
Comment (brunnedu) [reply darts/utils/multioutput.py:10]: That sounds like a good idea, I'll see what I can do 👍

**Pull request merged**